OrbitGl/ProcessManager.cpp

```diff
@@ -18,12 +18,15 @@ constexpr uint64_t kGrpcCallTimeoutMilliseconds = 1000;
 
 class ProcessManagerImpl final : public ProcessManager {
  public:
-  explicit ProcessManagerImpl(std::shared_ptr<grpc::Channel> channel,
+  explicit ProcessManagerImpl(const std::shared_ptr<grpc::Channel>& channel,
                               absl::Duration refresh_timeout);
 
   void SetProcessListUpdateListener(
       const std::function<void(ProcessManager*)>& listener) override;
   std::vector<ProcessInfo> GetProcessList() const override;
+  void GetProcessMemory(
+      uint32_t pid, uint64_t address, uint64_t size,
+      const std::function<void(std::string&&)>& callback) override;
```

Comment:
Is there a reason for having this callback? Maybe just return the string here - let the caller handle the rest. Callback makes things a bit less readable.

Reply:
You're totally right (a result of refactoring the initial implementation).